app/Http/Controllers/Api/EloquentController.php

```diff
@@ -18,12 +23,37 @@ public function replaceRelationship(RequestInterface $request)
         $key = $this->keyForRelationship($request->getRelationshipName());
 
         $store = $this->api()->getStore();
-        $relatedModels = [];
-        foreach($request->getDocument()->getResources()->getAll() as $resourceObject) {
-            $relatedModels[] = $store->findRecord($resourceObject->getIdentifier());
+
+        /** @var BelongsTo|HasMany|BelongsToMany|MorphToMany $relation */
+        $relation = $model->{$key}();
+        if(is_a($relation, BelongsTo::class)){
+            if($request->getDocument()->getData()) {
```

Comment:
I see a mix of styles here: `if(is_a(...))` and `if (is_a(...))`. We should get a linter in place to codify the intended style, and have CircleCI run it for us - and perhaps [pre-commit](http://pre-commit.com) as well.